core/trino-main/src/main/java/io/trino/type/UuidOperators.java

```diff
@@ -28,18 +33,62 @@
 import static io.trino.spi.function.OperatorType.CAST;
 import static io.trino.spi.type.UuidType.javaUuidToTrinoUuid;
 import static io.trino.spi.type.UuidType.trinoUuidToJavaUuid;
-import static java.util.UUID.randomUUID;
 
 public final class UuidOperators
 {
+    // All generators are thread safe
+    private static final RandomBasedGenerator V4_GENERATOR = Generators.randomBasedGenerator();
+    private static final TimeBasedReorderedGenerator V6_GENERATOR = Generators.timeBasedReorderedGenerator();
+    private static final TimeBasedEpochRandomGenerator V7_GENERATOR = Generators.timeBasedEpochRandomGenerator();
+
     private UuidOperators() {}
 
-    @Description("Generates a random UUID")
-    @ScalarFunction(deterministic = false)
+    @Description("Generates a random UUID v4 (RFC 4122)")
+    @ScalarFunction(deterministic = false, alias = "uuid_v4")
     @SqlType(StandardTypes.UUID)
     public static Slice uuid()
     {
-        java.util.UUID uuid = randomUUID();
+        java.util.UUID uuid = V4_GENERATOR.generate();
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v6 (RFC-9562)")
+    @ScalarFunction(deterministic = false)
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_v6()
+    {
+        java.util.UUID uuid = V6_GENERATOR.generate();
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v6 from a given timestamp (RFC-9562)")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    @LiteralParameters("u")
+    @Constraint(variable = "u", expression = "min(u, 6)")
+    public static Slice uuid_v6(@SqlType("timestamp(u) with time zone") long rawTimestamp)
+    {
+        java.util.UUID uuid = V6_GENERATOR.construct(rawTimestamp);
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v7 (RFC-9562)")
+    @ScalarFunction(deterministic = false)
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_v7()
+    {
+        java.util.UUID uuid = V7_GENERATOR.generate();
+        return javaUuidToTrinoUuid(uuid);
+    }
+
+    @Description("Generates a random UUID v7 from a given timestamp (RFC-9562)")
+    @ScalarFunction(deterministic = false)
+    @SqlType(StandardTypes.UUID)
+    @LiteralParameters("u")
+    @Constraint(variable = "u", expression = "min(u, 6)")
+    public static Slice uuid_v7(@SqlType("timestamp(u) with time zone") long rawTimestamp)
```

Comment:
According to the [RFC](https://www.rfc-editor.org/rfc/rfc9562.html#name-uuid-version-7), UUIDv7 is can have:

* An OPTIONAL sub-millisecond timestamp fraction (12 bits at maximum) as per [Section 6.2](https://www.rfc-editor.org/rfc/rfc9562.html#monotonicity_counters) (Method 3).
* An OPTIONAL carefully seeded counter as per [Section 6.2](https://www.rfc-editor.org/rfc/rfc9562.html#monotonicity_counters) (Method 1 or 2).
* Random data for each new UUIDv7 generated for any remaining space.

How does the implementation deal with those aspects? In particular, the "carefully seeded counter" in a distributed execution context.